ui/Screen/Wallet/Transactions.svelte

```diff
@@ -0,0 +1,64 @@
+<script lang="typescript">
+  import { store as transactions, getShortMonth } from "ui/src/transaction";
+
+  import TxList from "ui/DesignSystem/Component/Wallet/Transactions/TxList.svelte";
+
+  $: pendingTxs = $transactions.filter(
+    tx => tx.status === "Awaiting inclusion"
+  );
+  $: rejectedTxs = $transactions.filter(tx => tx.status === "Rejected");
+
+  $: txMonthSections = $transactions.reduce((res, cur) => {
```

Comment:
In general, using `reduce` is a good idea. But in this case an imperative loop has some advantages. It’s easier to typecheck, easier to understand, we don’t rely on some implicit ordering of the transactions, and it’s more performant. The latter is particularly important when dealing with lists because they quickly grow in size. I’d propose the following
```typescript
const groupTxs = (txs: Tx[]) => {
  const sections: Array<{ key: string; title: string; items: Tx[] }> = [];
  // Sort from newest to oldest
  txs.sort((a, b) => b.date - a.date);
  for (const tx of txs) {
    const txDate = new Date(tx.date);
    const txMonth = txDate.getMonth();
    const txYear = txDate.getFullYear();
    const key = `${txYear}-${txMonth}`;
    const currentSection = sections[sections.length - 1];
    if (currentSection && currentSection.key === key) {
      currentSection.items.push(tx);
    } else {
      const title = `${getShortMonth(txDate)} ${txYear}`;
      sections.push({
        key,
        title,
        items: [tx],
      });
    }
  }
  return sections;
};

$: txMonthSections = groupTxs($transactions);
```